src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,7 +120,8 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
+PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {_input.LA(1) == '.' || _input.LA(1) == '}'
+                     || Character.isWhitespace(_input.LA(1))}? {referenceCatched = true;};
```

Comment:
That's what I did, in a way. The thing is it will not be efficient to look ahead till the end of input in lexer to find a "non related symbol". So I just ensure that the token immediately following package is a related one. Anyway, the right syntax is to use `"` to let the lexer know it's a string. Also, our lexer can't identify `text`, it can just identify one `CHAR` at a time and the parser builds [`text`](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.g4#L919) out it.